app/Http/Controllers/MessageController.php

```diff
@@ -5,27 +5,43 @@
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\DB;
 use App\Models\Message;
+use Illuminate\Contracts\Database\Eloquent\Builder;
+use Illuminate\Support\Facades\Auth;
+use Symfony\Component\HttpFoundation\Response;
 
 class MessageController extends Controller
 {
     public function index(Request $request)
     {
-        $messages = DB::table('messages')->get();
-
-        return response()->json($messages);
+        $messages = Message::query()
+                ->leftjoin('users', 'user_uuid', '=', 'users.uuid')
+                ->select('messages.uuid', 'name', 'user_uuid', 'message', 'like_count', 'messages.created_at')
+                ->withExists('likes as like_status', fn (Builder $query) =>
+                    $query->where('user_uuid', Auth::id())
+                )->orderByDesc('created_at')->limit(100)->get();
```

Comment:
```suggestion
                )->orderByDesc('created_at')->limit(100)->toBase()->get();
```

JOIN することによってもはやこれはモデルでは無くなってしまっているので，そのニュアンスを示すために `->toBase()` を入れるとよいかもしれません。こうすると結果が `Illuminate\Database\Eloquent\Collection<Message>` ではなく `Illuminate\Support\Collection<stdClass>` になります

Reply:
toBase()を入れるとwithExistsのbool型へのキャストが剥がれてしまうようなので、一旦入れずに対処しようと思います。
対処法が見つかればその時修正します。

Reply:
あーなるほど！確かにそれは剥げますね
うーむ，非常に悩ましいけどここはこのままが正解か…